### DISCUSSION TITLE
sliding window horizon, data leakage

### DISCUSSION THREAD

**Discussion by georgemossad** [timeseriesAI/tsai#722] [Q&A] [answered]
hey everyone, I hope you are doing well  
How should I set the horizon parameter when applying a sliding window function to my multi-variate, multi-label data as I don't really get what it does? 

Additionally, what steps should I take to avoid leakage in my data? Should I split the training and test data and apply the sliding window function separately to each of them?
thanks for help

**Comment by oguiza** [comment, marked as answer]
I'd suggest you take a look at https://timeseriesai.github.io/tsai/data.preparation.html and try to understand how the sliding window works. There are many examples that might be helpful.

**Comment by georgemossad** [comment]
thanks, i would consider that.
